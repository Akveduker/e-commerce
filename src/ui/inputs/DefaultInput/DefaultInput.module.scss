@import "../../../styles/_mixins.scss";
.input {
    border-radius: $borderRadius;
    background-color: $C1H;
    border: 1px solid $C1D;
    width: 100%;
    padding: 11px 21px;
    font-size: 14px;
    &::placeholder {
        color: $C1C;
    }
}
.with-children {
    display: block;
    border-radius: $borderRadius;
    background-color: $C1H;
    border: 1px solid $C1D;
    width: 100%;
    overflow: hidden;
    position: relative;
    &__input {
        padding: 11px 21px;
        width: 100%;
        background-color: transparent;
        border: none;
        &::placeholder {
            color: $C1C;
        }
    }
    &__children {
        position: absolute;
        right: 10px;
        top: 0;
        bottom: 0;
        margin: auto 0;
    }
}
