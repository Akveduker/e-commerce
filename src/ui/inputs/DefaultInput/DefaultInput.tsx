import { FC } from 'react';
import InputMask from 'react-input-mask';
import { Props } from 'react-input-mask';
import { DefaultInputProps } from '../../../types/formInput/formInput';
import { mergeClassNames } from '../../../utils/class/mergeClassNames/mergeClassNames';
import s from './DefaultInput.module.scss'
const DefaultInput: FC<DefaultInputProps> = ({ childrenClassName, labelClassName, children, inputType, ...inputProps }) => {
    const childrenClass = mergeClassNames(s['with-children__children'], childrenClassName)
    const labelClass = mergeClassNames(s['with-children'], labelClassName)
    if (children) {
        return (
            <label className={labelClass} data-testid={'label'}>
                {inputType === 'mask' ?
                    <InputMask
                        className={s['with-children__input']}
                        {...inputProps as Props}
                    />
                    :
                    <input  {...inputProps as React.HTMLProps<HTMLInputElement>} className={s['with-children__input']} data-testid={'default'} />
                }
                <div className={childrenClass}>
                    {children}
                </div>
            </label>
        )
    }
    return (
        inputType === 'mask' ?
            <InputMask
                {...inputProps as Props}
                className={s['input']}
            />
            :
            <input  {...inputProps as React.HTMLProps<HTMLInputElement>} className={s['input']} data-testid={'default'} />

    );
};

export default DefaultInput;