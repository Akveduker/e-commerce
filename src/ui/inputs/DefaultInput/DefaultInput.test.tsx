import { render, screen } from "@testing-library/react"
import DefaultInput from "./DefaultInput"

describe('DefaultInput', () => {
    describe('С маской', () => {
        test('Рендерится нужный инпут, снапшот', () => {
            const { container } = render(<DefaultInput inputType='mask' mask={''} />)
            expect(screen.queryByTestId('label')).toBeNull()
            expect(screen.queryByTestId('default')).toBeNull()
            expect(container).toMatchSnapshot()
        })
        test('Добавляется доп элемент и работает связыванние классов', () => {
            const { container } = render(
                <DefaultInput inputType='mask' mask={''} childrenClassName={'children'} labelClassName={'label'}>
                    elem
                </DefaultInput>
            )
            expect(screen.getByTestId('label')).toHaveClass('label')
            const children = screen.queryByText('elem')
            expect(children).toBeInTheDocument()
            expect(children).toHaveClass('children')
            expect(container).toMatchSnapshot()
        })
    })
    describe('Без маски', () => {
        test('Рендерится нужный инпут, снапшот', () => {
            const { container } = render(<DefaultInput />)
            expect(screen.queryByTestId('label')).toBeNull()
            expect(screen.queryByTestId('default')).toBeInTheDocument()
            expect(container).toMatchSnapshot()
        })
        test('Добавляется доп элемент и работает связыванние классов', () => {
            const { container } = render(
                <DefaultInput childrenClassName={'children'} labelClassName={'label'}>
                    elem
                </DefaultInput>
            )
            expect(screen.getByTestId('label')).toHaveClass('label')
            const children = screen.queryByText('elem')
            expect(children).toBeInTheDocument()
            expect(children).toHaveClass('children')
            expect(container).toMatchSnapshot()
        })
    })
})