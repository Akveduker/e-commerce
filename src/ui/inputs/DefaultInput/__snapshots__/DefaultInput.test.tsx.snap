// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultInput Без маски Добавляется доп элемент и работает связыванние классов 1`] = `
<div>
  <label
    class="with-children label"
    data-testid="label"
  >
    <input
      class="with-children__input"
      data-testid="default"
    />
    <div
      class="with-children__children children"
    >
      elem
    </div>
  </label>
</div>
`;

exports[`DefaultInput Без маски Рендерится нужный инпут, снапшот 1`] = `
<div>
  <input
    class="input"
    data-testid="default"
  />
</div>
`;

exports[`DefaultInput С маской Добавляется доп элемент и работает связыванние классов 1`] = `
<div>
  <label
    class="with-children label"
    data-testid="label"
  >
    <input
      class="with-children__input"
    />
    <div
      class="with-children__children children"
    >
      elem
    </div>
  </label>
</div>
`;

exports[`DefaultInput С маской Рендерится нужный инпут, снапшот 1`] = `
<div>
  <input
    class="input"
  />
</div>
`;
